### PR TITLE
`Quiz exercises`: Improve performance by reducing unnecessary database calls related to quiz batches

### PR DESCRIPTION
**Pull request by bottbenj**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).

### Motivation and Context

Quiz batches were loaded a lot when they were not needed.

### Description

Dashboard used to load the batches for every synchronized quiz exercise, even tho it can only be useful when the quiz is currently active.
Beyond that there were a lot of other places were the batches were loaded with some other request even tho they were not required.

### Steps for Testing
Noting in particular should have changed, other than that in the query log there should be a lot less queries involving quiz batches.

Just checking over quiz things in general is a good idea tho.

#### Code Review
- [ ] Review 1
- [ ] Review 2
#### Manual Tests
- [ ] Test 1
- [ ] Test 2

### CODE REVIEW

File: src/main/java/de/tum/in/www1/artemis/repository/QuizExerciseRepository.java [hunk 1]
```diff
@@ -51,7 +50,7 @@ public interface QuizExerciseRepository extends JpaRepository<QuizExercise, Long
     @EntityGraph(type = LOAD, attributePaths = { "quizQuestions", "quizPointStatistic", "quizQuestions.quizQuestionStatistic", "categories", "quizBatches" })
     Optional<QuizExercise> findWithEagerQuestionsAndStatisticsById(Long quizExerciseId);
 
-    @EntityGraph(type = LOAD, attributePaths = { "quizQuestions", "quizBatches" })
+    @EntityGraph(type = LOAD, attributePaths = { "quizQuestions" })
     Optional<QuizExercise> findWithEagerQuestionsById(Long quizExerciseId);
```
Comment (manuelmanso) [inline src/main/java/de/tum/in/www1/artemis/repository/QuizExerciseRepository.java:54]: Shouldn't this method now be called something like "findWithEagerQuestionsAndBatchesById"?

File: src/main/java/de/tum/in/www1/artemis/repository/QuizExerciseRepository.java [hunk 2]
```diff
@@ -76,11 +75,6 @@ default QuizExercise findOne(Long quizExerciseId) {
      * @param quizExerciseId the id of the entity
      * @return the entity
      */
-    @Nullable
-    default QuizExercise findOneWithQuestions(Long quizExerciseId) {
-        return findWithEagerQuestionsById(quizExerciseId).orElse(null);
-    }
-
     @NotNull
     default QuizExercise findByIdWithQuestionsElseThrow(Long quizExerciseId) {
```
Comment (manuelmanso) [inline src/main/java/de/tum/in/www1/artemis/repository/QuizExerciseRepository.java:79]: Same as above, should change this method to  "findByIdWithQuestionsAndBatchesElseThrow"?

**Review by manuelmanso — APPROVED** [review]
The changes have been discussed and the performance improvements are clear.

**Review by SimonPaul02 — DISMISSED** [review]
Tested on TS1, works without errors. Nice job! (I couldn't measure the performance improvement, but it is pretty fast👌)

**Review by yuling-sun — APPROVED** [review]
Tested on TS1. Quiz exercises still work as expected. I didn't notice any other changes.

**Review by nadinesamyy — DISMISSED** [review]
tested quiz functionalities on ts2 and worked well

**Review by jbllhff — DISMISSED** [review]
Tested on ts2, works fine. It is not possible to import a quiz, is this intended?

**Comment by bottbenj** [comment]
> Tested on ts2, works fine. It is not possible to import a quiz, is this intended?

Importing full quiz exercises in another PR, importing is only possible one the question level here.

**Pull request merged**